### PR TITLE
Feature/hugo malhar

### PR DESCRIPTION
**Pull request by HugoTse**
- Added a markdown section explaining that we’re running the query of Alan Turing to get embeddings that capture the semantic meaning which we use to calculate the dot product to determine similarity between sentences.
- Added a conclusion and comments to explain that we loop over the indices of the top k relevant documents.
- Used the results of the semantic search as context to a prompt that helps the user visualize the model sticking to the provided data.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+    "doc_embeddings.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "GbYAXaI4RQiH"
+   },
+   "source": [
+    "### Step3: Embed query and compute dot product with document embeddings\n",
```
Comment (tugglep) [inline cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb:185]: add a space between Step and 3

File: cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+    "    print(\"Title:\", docs[doc_id][\"title\"])\n",
+    "    print(\"Text:\", docs[doc_id][\"text\"])\n",
+    "    print(docs[doc_id][\"url\"], \"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step4: Add the results of semantic search as context to a prompt\n",
```
Comment (tugglep) [inline cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb:337]: space between step and 4

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "GbYAXaI4RQiH"
+   },
+   "source": [
+    "### Step3: Embed query and compute dot product with document embeddings\n",
+    "We can now search these vectors for any query we want. For this toy example, we'll ask a question about Alan Turing since we know the Wikipedia page for Alan Turing is included in this subset of the archive.\n",
```
Comment (tugglep) [inline cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:186]: I would nix the word 'toy'

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+      "\n",
+      "After the war, Turing continued his work in computer science and mathematics. He designed the Automatic Computing Engine (ACE), one of the first designs for a \n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Send the same query to the Command R+ model using the Bedrock converse API but add the results of the semantic search as context\n",
+    "conversation = [\n",
```
Comment (tugglep) [inline cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:497]: I believe Command R models do not support conversation turns... please check my knowledge but if that's the case we may want to note that for the user.

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+    "]\n",
+    "\n",
+    "try:\n",
+    "    for i in range(3):\n",
+    "        print('Result of API call', str(i+1), ':')\n",
+    "        # Send the message to the model, using a basic inference configuration.\n",
+    "        response = bedrock_rt.converse(\n",
+    "            modelId='cohere.command-r-plus-v1:0',\n",
+    "            messages=conversation,\n",
+    "            inferenceConfig={\"maxTokens\": 200, \"temperature\": 0.5, \"topP\": 0.9},\n",
```
Comment (tugglep) [inline cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:511]: Should we note why we're using temp = .5

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 4 above)
Comment (breanne-warner) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:497]: this is correct does not support multi conversation turns on Bedrock

File: cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb (hunk 1 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb:185]: Added a space between "Step" and "3".

File: cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb (hunk 2 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/.ipynb_checkpoints/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives-checkpoint.ipynb:337]: Added a space between "Step" and "4".

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 3 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:186]: Deleted "toy".

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 5 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:511]: Added a note to explain that 0.5 is used to illustrate variance in the response. Linked the Amazon Bedrock User Guide for readers interested in changing these values.

File: cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb (hunk 4 above)
Comment (HugoTse) [reply cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb:504]: Got it! I added a note clarifying that the user needs to use the Command R model (not the Command R+ model) if they want to include a chat_history/conversation with multiple turns.

**Pull request merged**